common/src/main/java/com/wynntils/features/inventory/HightlightDuplicateCosmeticsFeature.java

```diff
@@ -0,0 +1,151 @@
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.inventory.ChestMenu;
+import net.minecraft.world.inventory.Slot;
+import net.minecraft.world.item.Items;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@ConfigCategory(Category.INVENTORY)
+public class HightlightDuplicateCosmeticsFeature extends Feature {
+    @Persisted
+    public final Config<HighlightCondition> highlightCondition = new Config<>(HighlightCondition.BOTH);
+
+    @Persisted
+    public final Config<CustomColor> hoveredHighlightColor = new Config<>(CommonColors.BLUE);
+
+    @Persisted
+    public final Config<CustomColor> selectedHighlightColor = new Config<>(CommonColors.ORANGE);
+
+    private static final Component ADD_REWARD_TEXT = Component.literal("§7Click on a reward to add it");
+    private static final Component RETURN_TEXT = Component.literal("§7Return to Scrap Menu");
+    private static final int RETURN_SLOT = 18;
+    private static final List<Integer> SELECTED_COSMETIC_SLOTS = List.of(1, 2, 3, 4, 5);
+
+    private final List<Component> selectedCosmetics = new ArrayList<>();
+
+    private boolean onScrapMenu = false;
+    private Component hoveredCosmetic;
+
+    @SubscribeEvent
+    public void onScreenInit(ScreenInitEvent event) {
+        if (!(event.getScreen() instanceof AbstractContainerScreen<?> screen)) return;
+        if (!(screen.getMenu() instanceof ChestMenu)) return;
+
+        StyledText title = StyledText.fromComponent(screen.getTitle());
+
+        SearchableContainerType searchableContainerType = SearchableContainerType.getContainerType(title);
+        if (searchableContainerType != SearchableContainerType.SCRAP_MENU) return;
+
+        onScrapMenu = true;
+    }
+
+    @SubscribeEvent
+    public void onScreenClose(ScreenClosedEvent e) {
+        selectedCosmetics.clear();
+        onScrapMenu = false;
+        hoveredCosmetic = null;
+    }
+
+    @SubscribeEvent
+    public void onContainerRender(ContainerRenderEvent event) {
+        if (!onScrapMenu) return;
+        // No need to get hovered cosmetic if only displaying selected cosmetics
+        if (highlightCondition.get() == HighlightCondition.SELECTED) return;
+
+        Slot hoveredSlot = event.getHoveredSlot();
+
+        // If no item or on new reward screen don't get a hovered cosmetic, hide §f too as the glass panes on new
+        // reward screen are named that and the return text item isn't added immediately
+        if (hoveredSlot == null
+                || hoveredSlot.getItem().getItem() == Items.AIR
+                || event.getScreen()
+                        .getMenu()
+                        .slots
+                        .get(RETURN_SLOT)
+                        .getItem()
+                        .getHoverName()
+                        .equals(RETURN_TEXT)
+                || hoveredSlot.getItem().getHoverName().equals(Component.literal("§f"))) {
+            hoveredCosmetic = null;
+            return;
+        }
```

Comment:
1. The comment is a bit weird to read to me, might just be me though
2. Please clear this if condition up. It's long.
3. Extract `Component.literal("§f")` as a const?